Mathlib/Data/Fin/Basic.lean

```diff
@@ -1580,6 +1592,14 @@ theorem one_succAbove_one {n : ℕ} : (1 : Fin (n + 3)).succAbove 1 = 2 := by
   exact this
 #align fin.one_succ_above_one Fin.one_succAbove_one
 
+lemma rev_succAbove (p : Fin (n + 1)) (i : Fin n) :
```

Comment:
This lemma looks like it might be the wrong way round?

Reply:
Probably, you're right. Feel free to change it.

Reply:
I wonder what the general property of a function that commutes with succAbove in this way is?